### ISSUE TITLE
Bolt-on methods, prototypes and more

### ISSUE THREAD

**Issue by ghost** [ractivejs/ractive#3129]
## Description:

TL;DR: If Ractive can't shed weight by code removal/tree-shaking/exercise and diet, at least it can shed weight by opting-in to features.

Mostly inspired by the way RxJS splits out operators. RxJS has a fully bundled build, containing everything under the `Rx` namespace. Great when importing the entire library or if you just want the "batteries included" experience. But it's not so great when it comes to size.

But if you have a build-step, RxJS also [provides the library _by parts_](https://github.com/ReactiveX/rxjs#es6-via-npm) (see section that mentions "size-sensitive bundling"). You can essentially bolt on methods, properties, etc. using imports, only add what you need and leave out what you don't.

For instance, the following code imports all 200KB of Ractive goodness even if you only need `Ractive.VERSION`, `Ractive.extend` and `Ractive.prototype.set`.

```
import Ractive from 'ractive

const Component = Ractive.extend({
  template: `
    {{ message }}
  `
  onrender(){
    console.log(Ractive.VERSION)
  }
})

const instance = Component({ el: 'body' })

instance.set('message', 'Hello, World!')
```

But in the following, it would only import `Ractive.extend`, `Ractive.VERSION`, and `Ractive.prototype.set`. `Ractive.extend` would return an instantiable constructor, but when instantiated from, the instance will only contain `Ractive.prototype.set`. All the other methods are not included and you try to call one that's not there, it will throw.

```
import {extend, VERSION} from '@ractivejs/core'
import '@ractivejs/core/methods/set'

const Component = extend({
  template: `
    {{ message }}
  `
  onrender(){
    console.log(VERSION)
  }
})

const instance = Component({ el: 'body' })

instance.set('message', 'Hello, World!')

// Would fail since you didn't import '@ractivejs/core/methods/teardown'
// But the entire teardown code wouldn't be added to the bundle at all.
instance.teardown() 
```

Another advantage is, since it declares dependencies as actual imports, tools like Rollup can actually tree-shake the code better.

This would also require a rewrite since Ractive is very monolithic. And again, this would just be another option of using Ractive. The 200kb, one-size-fits-all build will still be there. Also, mileage may vary since some internal workings of Ractive actually call the same APIs we're trying to shed off. So they might be bundled regardless if the consumer added them or not.
